tokio/src/sync/semaphore.rs

```diff
@@ -80,6 +80,10 @@ impl Semaphore {
     }
 
     /// Adds `n` new permits to the semaphore.
+    ///
+    /// `n` must less than [`MAX_PERMITS`]
+    ///
+    /// [`MAX_PERMITS`]: Semaphore::MAX_PERMITS
     pub fn add_permits(&self, n: usize) {
```

Comment:
The link does not work as `MAX_PERMITS` is not public. We should write the actual number instead.

We should also avoid starting a sentence with \`n\`. Perhaps something like "The maximum number of permits is N. This function will panic if this number is exceeded."

Reply:
```rust
    /// Adds `n` new permits to the semaphore.
    ///
    /// The maximum number of `n` is  2305843009213693951 (`std::usize::MAX >> 3`).
    ///
    /// And this function will panic if this number is exceeded.
```

Reply:
The limit is on the total number of permits, so it can panic even with `n = 10` as in [this example][1]. I would like to catch this nuance in the documentation.

[1]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9eeaab3b04d24e954509dbdb64c445b1

Reply:
````
    /// Adds `n` new permits to the semaphore.
    ///
    /// The maximum number of `n` must less than 2305843009213693951 - `the initial number of permits`.
    ///
    /// And this function will panic if this number is exceeded.
````
How about this.

Reply:
I think it would be simpler to just say

> The maximum number of permits is `usize::MAX >> 3`, and this function will panic if the limit is exceeded.

Reply:
Ok, got it. My English is poor, sorry.

Reply:
I've commit the new version of the document